### PR TITLE
[dtensor][debug] add all_reduce_coalesced tracing to CommDebugMode

### PR DESCRIPTION
**Pull request by sinhaanshul**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #127040
* #127029
* __->__ #127025

**Summary**
Added all_reduce_coalesced tracing to CommDebugMode and added test case to test_comm_mode test suite.

**Test Plan**
pytest test/distributed/_tensor/debug/test_comm_mode.py 

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang @d4l3k

### CODE REVIEW

**Comment by sinhaanshul** [comment]
Abandoned old #126928 because of messed up git tree

File: test/distributed/_tensor/debug/test_comm_mode.py [hunk 1]
```diff
@@ -59,6 +59,46 @@ def forward(self, x):
...
+            model(torch.randn(20, 10, device=self.device_type))
+            model2(torch.randn(20, 10, device=self.device_type))
+
+        comm_counts = comm_mode.get_comm_counts()
+        self.assertEqual(comm_mode.get_total_counts(), 6)
+        self.assertEqual(comm_counts[c10d_functional.all_reduce_coalesced], 1)
+        self.assertEqual(comm_counts[c10d_functional.all_reduce], 1)
+        self.assertEqual(comm_counts[c10d_functional.all_gather_into_tensor], 2)
+        self.assertEqual(comm_counts[c10d_functional.reduce_scatter_tensor], 2)
+
```
Comment (XilunWu) [inline test/distributed/_tensor/debug/test_comm_mode.py:101]: Let's organize the test in this format:
```

**Review by XilunWu — COMMENTED** [review]
let's re-organize the test

File: test/distributed/_tensor/debug/test_comm_mode.py (hunk 1 above)
Comment (XilunWu) [reply test/distributed/_tensor/debug/test_comm_mode.py:108]: this test has repeated part. fixed it in next PR.

**Review by XilunWu — APPROVED** [review]

**Pull request closed without merging**